scipy/stats/_binomtest.py

```diff
@@ -0,0 +1,258 @@
+    """
+    Perform a test that the probability of success is p.
+
+    This is a test of the null hypothesis that the probability of success
+    in a Bernoulli experiment is `p`.
+
+    Parameters
+    ----------
+    k : int or array_like
+        The number of successes, or if k has length 2, it is the
+        number of successes and the number of failures.
+    n : int
+        The number of trials.  This is ignored if k gives both the
+        number of successes and failures.
+    p : float, optional
+        The hypothesized probability of success.  ``0 <= p <= 1``. The
+        default value is ``p = 0.5``.
+    alternative : {'two-sided', 'greater', 'less'}, optional
+        Indicates the alternative hypothesis. The default value is
+        'two-sided'.
+
+    Returns
+    -------
+    result : BinomTestResult instance
+        The return value is an object with the following attributes:
+
+        * `k` : int
+        * `n` : int
+        * `alternative` : str
+        * `pvalue` : float
+        * `proportion_estimate` : float
+
+        The object has the following methods:
+
+        * `proportion_ci(confidence_level=0.95, method='exact')`
+            Compute the confidence interval for ``proportion_estimate``.
+
+    References
+    ----------
+    .. [1] https://en.wikipedia.org/wiki/Binomial_test
+
+    Examples
+    --------
+    >>> from scipy.stats import binomtest
+
+    A car manufacturer claims that no more than 10% of their cars are unsafe.
+    15 cars are inspected for safety, 3 were found to be unsafe. Test the
+    manufacturer's claim:
+
+    >>> result = binomtest(3, n=15, p=0.1, alternative='greater')
+    >>> result.pvalue
+    0.18406106910639114
+
+    The null hypothesis cannot be rejected at the 5% level of significance
+    because the returned p-value is greater than the critical value of 5%.
+
+    The estimated proportion is simply ``3/15``:
+
+    >>> result.proportion_estimate
+    0.2
+
+    We can use the `proportion_ci()` method of the result to compute the
+    confidence interval of the estimate:
+
+    >>> result.proportion_ci(confidence_level=0.95)
+    ConfidenceInterval(low=0.056846867590246826, high=1.0)
+
+    """
+    k = np.atleast_1d(k).astype(np.int_)
+    if len(k) == 2:
+        n = k[1] + k[0]
+        k = k[0]
+    elif len(k) == 1:
+        k = k[0]
+        if n is None or n < k:
+            raise ValueError("n must be >= k")
+        n = np.int_(n)
+    else:
+        raise ValueError("Incorrect length for k.")
+
+    if (p > 1.0) or (p < 0.0):
+        raise ValueError("p must be in range [0,1]")
+
+    if alternative not in ('two-sided', 'less', 'greater'):
+        raise ValueError("alternative not recognized\n"
+                         "must be 'two-sided', 'less' or 'greater'")
+
+    if alternative == 'less':
+        pval = binom.cdf(k, n, p)
+    elif alternative == 'greater':
+        pval = binom.sf(k-1, n, p)
+    else:
+        # alternative is 'two-sided'
```

Comment:
Is there a reference for how a two-sided test should be handled?

Reply:
I added references in my latest update.

Reply:
OK. I don't have that book, and I would prefer not to hunt down the information, so I will not be checking this for accuracy (other than comparing output against other software).

Reply:
There are some alternate cases, e.g. ***randomized*** tests and ***mid pvalue*** methods that could be done. 
I don't want to bikeshed and add those into this PR though so just a comment in passing that if someone wanted to add those in future we might consider sticking w/these methods and also accept a `callable` that does the calculation for other methods besides these 3. 

Supporting something like that might require a little (but not much) reworking.